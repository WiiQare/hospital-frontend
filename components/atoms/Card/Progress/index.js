const CardProgress = ({ color, value, title, amount, className }) => {
  return (
    <div
      className={`${className} py-4 px-4 rounded-lg flex items-center gap-4`}
    >
      <div
        className={`radial-progress ${color}`}
        style={{ "--value": value, "--size": "4rem" }}
      >
        {value}%
      </div>
      <div>
        <h3 className="text-gray-700 font-bold text-sm">{title}</h3>
        <span className="text-sm">${amount}</span>
      </div>
    </div>
  );
};

export default CardProgress;
