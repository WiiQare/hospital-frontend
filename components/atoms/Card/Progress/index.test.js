import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Progress from ".";

describe("Progress", () => {
  const value = 50;
  let component;
  beforeEach(() => {
    const res = render(<Progress value={value} />);
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render the value", () => {
    expect(screen.getByText(`${value}%`)).toBeInTheDocument();
  });
});
