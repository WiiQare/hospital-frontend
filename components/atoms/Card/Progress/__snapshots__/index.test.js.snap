// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render the component 1`] = `
<div>
  <div
    class="undefined py-4 px-4 rounded-lg flex items-center gap-4"
  >
    <div
      class="radial-progress undefined"
      style="--value: 50; --size: 4rem;"
    >
      50
      %
    </div>
    <div>
      <h3
        class="text-gray-700 font-bold text-sm"
      />
      <span
        class="text-sm"
      >
        $
      </span>
    </div>
  </div>
</div>
`;
